packages/stylelint-plugin/src/rules/no-invalid-equations.ts

```diff
@@ -0,0 +1,165 @@
+            negated: false,
+          })})`
+          node.value = fixedVariable
+          amended = postcssValueParser.stringify(parsed.nodes)
+        } else {
+          options.reporter({
+            message: negatedKaizenVariableMessage,
+            node: postcssNode,
+            autofixAvailable: !isVariable(postcssNode),
+          })
+        }
+
+        return false
+      }
+
+      lastTwoWordNodesNextToEachOther.unshift(node)
+      if (lastTwoWordNodesNextToEachOther.length === 3) {
+        lastTwoWordNodesNextToEachOther.pop()
+      }
+      if (lastTwoWordNodesNextToEachOther.length === 2) {
+        const parsedVariable0 = parseVariable(
+          lastTwoWordNodesNextToEachOther[0]
+        )
+        const parsedVariable1 = parseVariable(
+          lastTwoWordNodesNextToEachOther[1]
+        )
+
+        // If there exists an operator and a variable next to each other, it's invalid
+        if (
+          (isOperator(lastTwoWordNodesNextToEachOther[0].value) &&
+            parsedVariable1?.kaizenToken) ||
+          (isOperator(lastTwoWordNodesNextToEachOther[1].value) &&
+            parsedVariable0?.kaizenToken)
+        ) {
+          const variableInEquation = parsedVariable0 || parsedVariable1
+          if (isWithinCalcFunction) {
+            // If the variable is interpolated within a calc function, its fine
+            if (variableInEquation?.interpolated) {
+              return false
+            }
+
+            if (
+              options.language === "scss" &&
+              options.fix &&
+              !isVariable(postcssNode)
+            ) {
+              if (variableInEquation?.kaizenToken) {
+                variableInEquation.node.value = stringifyVariable({
+                  ...variableInEquation,
+                  interpolated: true,
+                })
+              }
+
+              amended = postcssValueParser.stringify(parsed.nodes)
+            } else {
+              options.reporter({
+                message: tokenNotInterpolatedInCalcMessage,
+                node: postcssNode,
+                autofixAvailable: !isVariable(postcssNode),
+              })
+            }
+          } else {
+            options.reporter({
+              message: kaizenVariableUsedNextToOperatorMessage,
+              node: postcssNode,
+              autofixAvailable: false,
+            })
+          }
+
+          // Return false to break the walk
+          return false
+        }
+      }
+    }
+  })
+
+  if (options.fix && amended && !isVariable(postcssNode)) {
+    postcssNode.replaceWith(
+      postcssNode.clone({
+        value: amended,
+      })
+    )
+  }
+}
+
+export const noInvalidEquationsRule = (
+  stylesheetNode: Root,
+  options: Options
+) => {
+  walkDeclsWithKaizenTokens(stylesheetNode, ({ postcssNode, parsedValue }) => {
+    noInvalidEquationsOnDeclaraion(postcssNode, parsedValue, options)
+  })
+}
+
+export const declContainsInvalidEquations = (
```

Comment:
Again, I feel this export from one rule, being used in the `no-deprecated-tokens` rule, is a bit of a code-smell. I don't think rules should be importing utility functions from each other

Reply:
Following up in https://github.com/cultureamp/kaizen-design-system/pull/1624